### PR TITLE
fix: update celery and libsass versions

### PR DESCRIPTION
**Pull request by Faraz32123**
celery version needs to updated due to below error
```
1.512   File "/openedx/ecommerce/ecommerce/extensions/offer/utils.py", line 15, in <module>
1.513     from ecommerce_worker.email.v1.api import send_offer_assignment_email, send_offer_update_email
1.513   File "/openedx/venv/lib/python3.12/site-packages/ecommerce_worker/email/v1/api.py", line 7, in <module>
1.513     from .tasks import (
1.513   File "/openedx/venv/lib/python3.12/site-packages/ecommerce_worker/email/v1/tasks.py", line 6, in <module>
1.514     from celery import shared_task
1.514   File "/openedx/venv/lib/python3.12/site-packages/celery/__init__.py", line 19, in <module>
1.514     from . import local  # noqa
1.514     ^^^^^^^^^^^^^^^^^^^
1.514   File "/openedx/venv/lib/python3.12/site-packages/celery/local.py", line 17, in <module>
1.515     from .five import PY3, bytes_if_py2, items, string, string_t
1.515   File "/openedx/venv/lib/python3.12/site-packages/celery/five.py", line 7, in <module>
1.515     import vine.five
1.515   File "/openedx/venv/lib/python3.12/site-packages/vine/__init__.py", line 8, in <module>
1.515     from .abstract import Thenable
1.515   File "/openedx/venv/lib/python3.12/site-packages/vine/abstract.py", line 6, in <module>
1.516     from .five import with_metaclass, Callable
1.516   File "/openedx/venv/lib/python3.12/site-packages/vine/five.py", line 364, in <module>
1.516     from inspect import formatargspec, getargspec as _getargspec  # noqa
1.516     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
1.517 ImportError: cannot import name 'formatargspec' from 'inspect' (/opt/pyenv/versions/3.12.2/lib/python3.12/inspect.py). Did you mean: 'formatargvalues'?
```
libsass version needs to updated due to below error
```
1.603   File "/openedx/ecommerce/ecommerce/theming/management/commands/update_assets.py", line 7, in <module>
1.604     import sass
1.604   File "/openedx/venv/lib/python3.12/site-packages/sass.py", line 609, in <module>
1.604     class SassMap(collections.Mapping):
1.604                   ^^^^^^^^^^^^^^^^^^^
1.604 AttributeError: module 'collections' has no attribute 'Mapping'
```
This PR needs to be merged to close this tutor-ecommerce [issue](https://github.com/overhangio/tutor-ecommerce/issues/68). These changes work with python 3.8.0 to 3.12.2.

### CODE REVIEW

File: requirements/docs.txt [hunk 1]
```diff
@@ -1,24 +1,24 @@
 #
-# This file is autogenerated by pip-compile with Python 3.8
+# This file is autogenerated by pip-compile with Python 3.9
```
Comment (DawoudSheraz) [inline requirements/docs.txt]: we should still use Python 3.8 venv to run `make upgrade`, after adding the constraints, to ensure the dependencies work on 3.8 until redwood.

File: ecommerce/extensions/api/v2/urls.py [hunk 2]
```diff
@@ -160,15 +160,19 @@
...
-              basename='partner-catalogs', parents_query_lookups=['partner_id'])
-router.register(r'partners', partner_views.PartnerViewSet) \
-    .register(r'products', product_views.ProductViewSet,
-              basename='partner-product', parents_query_lookups=['stockrecords__partner_id'])
+partner_router = SimpleRouter()
+partner_router.register(r'partners', partner_views.PartnerViewSet, basename='partner')
+catalog_router = SimpleRouter()
+catalog_router.register(r'catalogs', catalog_views.CatalogViewSet, basename='partner-catalogs')
+product_router = SimpleRouter()
+product_router.register(r'products', product_views.ProductViewSet, basename='partner-product')
```
Comment (DawoudSheraz) [inline ecommerce/extensions/api/v2/urls.py]: query lookups are missing here.

Comment (Faraz32123) [reply ecommerce/extensions/api/v2/urls.py]: OH my bad, i missed those. let me add those.

File: requirements/docs.txt (hunk 1 above)
Comment (Faraz32123) [reply requirements/docs.txt]: done.

File: .github/workflows/migrations-mysql8.yml [hunk 3]
```diff
@@ -52,7 +52,7 @@ jobs:
         pip uninstall -y mysqlclient
         pip install --no-binary mysqlclient mysqlclient
         pip uninstall -y xmlsec
-        pip install --no-binary xmlsec xmlsec
+        pip install --no-binary xmlsec==1.3.13 xmlsec==1.3.13
```
Comment (DawoudSheraz) [inline .github/workflows/migrations-mysql8.yml:58]: nit: we should add a comment here to explain why this is needed, alongside the link to the issue.

File: requirements/constraints.txt [hunk 4]
```diff
@@ -31,7 +31,8 @@ pluggy<1.0.0  # pluggy==1.0.0 requires tox>3.14.6
 idna==2.7
 
 # TODO : Pinning this until we are sure there aren't any breaking changes, then we'll upgrade.
-celery<5.0.0
+celery<6.0.0
+libsass==0.23.0
```
Comment (DawoudSheraz) [inline requirements/constraints.txt]: Please add a comment to specify why are we pinning to 0.23.0 specifically.

**Review by DawoudSheraz — APPROVED** [review]
1-2 small comments to address before merge. CI is green, so 👍🏽

**Comment by DawoudSheraz** [comment]
@feanil Hi. Can you help here in getting a review from Axim, please? Thanks

**Comment by andrey-canon** [comment]
Hi @DawoudSheraz, @feanil I will make a review of this next week :D

**Review by andrey-canon — CHANGES_REQUESTED** [review]
I tested this branch in addition with https://github.com/overhangio/tutor-ecommerce/pull/72 and I didn't make a whole test  but the application loads  and the partner application(That was the unique application whose url definition changed)worked as expected, however,  when I tried some commands from the makefile  I got the following errors

`make requirements`

image

looks like the transifex-client dependency is not compatible with python 3.12

`make requirements.tox`  This command is executed by the `make static` command

This makes a downgrade of pip https://github.com/openedx/ecommerce/blob/master/Makefile#L48 and that generates the following error:
image

Could you fix that ?

**Comment by Faraz32123** [comment]
Hello @andrey-canon, thanks for your time.
I have fixed `transifex-client` compatibility issue with python3.12. The other issue will be fixed in #4153 and to test that PR related to CI, this PR needs to b merged.

**Review by andrey-canon — APPROVED** [review]
Everything looks okay. I tested the Transifex issue, and it was fixed. However, we cannot ensure that 100% of the application is working unless we run tests. To achieve that, we need to rebase this  [pr](https://github.com/openedx/ecommerce/pull/4153) onto this one. That's why I approve this under the premise that there are some current issues, which will be addressed in subsequent PRs.

CC @feanil

**Review by feanil — APPROVED** [review]

**Pull request merged**